frontend/src/actions/FiltersActions.js

```diff
@@ -1,8 +1,355 @@
-import { SET_CLEAR_ALL_FILTER } from '../constants/FilterTypes';
+import * as types from '../constants/FilterTypes';
+import deepUnfreeze from 'deep-unfreeze';
+import { fieldValueToString } from '../utils/tableHelpers';
+import _ from 'lodash';
 
-export function clearAllFilters(data) {
+export function clearAllFilters({ id, data }) {
   return {
-    type: SET_CLEAR_ALL_FILTER,
-    payload: data,
+    type: types.CLEAR_ALL_FILTERS,
+    payload: { id, data },
   };
 }
+
+/**
+ * @method createFilter
+ * @summary Add a new filter entry to the redux store
+ */
+export function createFilter({ id, data }) {
+  return {
+    type: types.CREATE_FILTER,
+    payload: { id, data },
+  };
+}
+
+/**
+ * @method deleteFilter
+ * @summary Remove the filter with specified `id` from the store
+ */
+export function deleteFilter(id) {
+  return {
+    type: types.DELETE_FILTER,
+    payload: { id },
+  };
+}
+
+/**
+ * @method updateNotValidFields
+ * @summary updates in the store the notValidFields flag
+ */
+export function updateNotValidFields({ filterId, data }) {
+  return {
+    type: types.UPDATE_FLAG_NOTVALIDFIELDS,
+    payload: { filterId, data },
+  };
+}
+
+/**
+ * @method updateActiveFilter
+ * @summary Updates the activeFilter in the store for the corresponding entity id
+ */
+export function updateActiveFilter({ id, data }) {
+  return {
+    type: types.UPDATE_ACTIVE_FILTER,
+    payload: { id, data },
+  };
+}
+
+/**
+ * @method updateInlineFilter
+ * @summary Updates the InlineFilter value in the store for the corresponding entity id
+ */
+export function updateInlineFilter({ filterId, data }) {
+  return {
+    type: types.UPDATE_INLINE_FILTER,
+    payload: { filterId, data },
+  };
+}
+
+/**
+ * @method updateFilterWidgetShown
+ * @summary Updates the widgetShown in the store for the corresponding entity id with a boolean value
+ */
+export function updateFilterWidgetShown({ id, data }) {
+  return {
+    type: types.FILTER_UPDATE_WIDGET_SHOWN,
+    payload: { id, data },
+  };
+}
+
+/**
+ * @method clearStaticFilters
+ * @summary Clears the existing static filters for a filter branch in the redux store
+ */
+export function clearStaticFilters({ filterId, data }) {
```

Comment:
Any particular reason why it's sometimes `id`, and other times `filterId` ? It's a minor thing but if we're already using named params let's stick to one convention.

Reply:
Agree. Updated to filterId all over. Thanks